# Borrador Facundo 

In [1]:
#Importacion de librerias/módulos
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

In [2]:
#Establezco la dirección del archivo
file_location='properatti.csv'

#Leo el data set
datos_crudos=pd.read_csv(file_location)

#Visualizo las primeras filas
display(datos_crudos.head(3))


Unnamed: 0 operation property_type place_name  \
0           0      sell            PH  Mataderos   
1           1      sell     apartment   La Plata   
2           2      sell     apartment  Mataderos   

                       place_with_parent_names country_name  \
0        |Argentina|Capital Federal|Mataderos|    Argentina   
1  |Argentina|Bs.As. G.B.A. Zona Sur|La Plata|    Argentina   
2        |Argentina|Capital Federal|Mataderos|    Argentina   

               state_name  geonames_id                  lat-lon        lat  \
0         Capital Federal    3430787.0  -34.6618237,-58.5088387 -34.661824   
1  Bs.As. G.B.A. Zona Sur    3432039.0  -34.9038831,-57.9643295 -34.903883   
2         Capital Federal    3430787.0  -34.6522615,-58.5229825 -34.652262   

   ...  surface_covered_in_m2  price_usd_per_m2 price_per_m2  floor  rooms  \
0  ...                   40.0       1127.272727  1550.000000    NaN    NaN   
1  ...                    NaN               NaN          NaN    NaN    NaN   
2  ...                   55.0       1309.090909  1309.090909    NaN    NaN   

   expenses                                      properati_url  \
0       NaN  http://www.properati.com.ar/15bo8_venta_ph_mat...   
1       NaN  http://www.properati.com.ar/15bob_venta_depart...   
2       NaN  http://www.properati.com.ar/15bod_venta_depart...   

                                         description  \
0  2 AMBIENTES TIPO CASA PLANTA BAJA POR PASILLO,...   
1  Venta de departamento en décimo piso al frente...   
2  2 AMBIENTES 3ER PISO LATERAL LIVING COMEDOR AM...   

                                               title  \
0                 2 AMB TIPO CASA SIN EXPENSAS EN PB   
1  VENTA Depto 2 dorm. a estrenar 7 e/ 36 y 37   ...   
2           2 AMB 3ER PISO CON ASCENSOR APTO CREDITO   

                                     image_thumbnail  
0  https://thumbs4.properati.com/8/BluUYiHJLhgIIK...  
1  https://thumbs4.properati.com/7/ikpVBu2ztHA7jv...  
2  https://thumbs4.properati.com/5/SXKr34F_IwG3W_...  

[3 rows x 26 columns]

In [3]:
#primero hacemos drop de las columnas operation y country_name
datos= datos_crudos.drop(['Unnamed: 0','operation','country_name'], axis=1)

datos.head(3)

property_type place_name                      place_with_parent_names  \
0            PH  Mataderos        |Argentina|Capital Federal|Mataderos|   
1     apartment   La Plata  |Argentina|Bs.As. G.B.A. Zona Sur|La Plata|   
2     apartment  Mataderos        |Argentina|Capital Federal|Mataderos|   

               state_name  geonames_id                  lat-lon        lat  \
0         Capital Federal    3430787.0  -34.6618237,-58.5088387 -34.661824   
1  Bs.As. G.B.A. Zona Sur    3432039.0  -34.9038831,-57.9643295 -34.903883   
2         Capital Federal    3430787.0  -34.6522615,-58.5229825 -34.652262   

         lon     price currency  ...  surface_covered_in_m2  price_usd_per_m2  \
0 -58.508839   62000.0      USD  ...                   40.0       1127.272727   
1 -57.964330  150000.0      USD  ...                    NaN               NaN   
2 -58.522982   72000.0      USD  ...                   55.0       1309.090909   

   price_per_m2  floor  rooms  expenses  \
0   1550.000000    NaN    NaN       NaN   
1           NaN    NaN    NaN       NaN   
2   1309.090909    NaN    NaN       NaN   

                                       properati_url  \
0  http://www.properati.com.ar/15bo8_venta_ph_mat...   
1  http://www.properati.com.ar/15bob_venta_depart...   
2  http://www.properati.com.ar/15bod_venta_depart...   

                                         description  \
0  2 AMBIENTES TIPO CASA PLANTA BAJA POR PASILLO,...   
1  Venta de departamento en décimo piso al frente...   
2  2 AMBIENTES 3ER PISO LATERAL LIVING COMEDOR AM...   

                                               title  \
0                 2 AMB TIPO CASA SIN EXPENSAS EN PB   
1  VENTA Depto 2 dorm. a estrenar 7 e/ 36 y 37   ...   
2           2 AMB 3ER PISO CON ASCENSOR APTO CREDITO   

                                     image_thumbnail  
0  https://thumbs4.properati.com/8/BluUYiHJLhgIIK...  
1  https://thumbs4.properati.com/7/ikpVBu2ztHA7jv...  
2  https://thumbs4.properati.com/5/SXKr34F_IwG3W_...  

[3 rows x 23 columns]

In [4]:
datos.columns

Index(['property_type', 'place_name', 'place_with_parent_names', 'state_name',
       'geonames_id', 'lat-lon', 'lat', 'lon', 'price', 'currency',
       'price_aprox_local_currency', 'price_aprox_usd', 'surface_total_in_m2',
       'surface_covered_in_m2', 'price_usd_per_m2', 'price_per_m2', 'floor',
       'rooms', 'expenses', 'properati_url', 'description', 'title',
       'image_thumbnail'],
      dtype='object')

In [5]:
print(datos)

       property_type       place_name  \
0                 PH        Mataderos   
1          apartment         La Plata   
2          apartment        Mataderos   
3                 PH          Liniers   
4          apartment           Centro   
...              ...              ...   
121215     apartment         Belgrano   
121216         house           Beccar   
121217     apartment    Villa Urquiza   
121218     apartment      Plaza Colón   
121219     apartment  Capital Federal   

                                  place_with_parent_names  \
0                   |Argentina|Capital Federal|Mataderos|   
1             |Argentina|Bs.As. G.B.A. Zona Sur|La Plata|   
2                   |Argentina|Capital Federal|Mataderos|   
3                     |Argentina|Capital Federal|Liniers|   
4       |Argentina|Buenos Aires Costa Atlántica|Mar de...   
...                                                   ...   
121215               |Argentina|Capital Federal|Belgrano|   
121216  |Argentina|

In [6]:
datos.isna().sum()

property_type                      0
place_name                        23
place_with_parent_names            0
state_name                         0
geonames_id                    18717
lat-lon                        51550
lat                            51550
lon                            51550
price                          20410
currency                       20411
price_aprox_local_currency     20410
price_aprox_usd                20410
surface_total_in_m2            39328
surface_covered_in_m2          19907
price_usd_per_m2               52603
price_per_m2                   33562
floor                         113321
rooms                          73830
expenses                      106958
properati_url                      0
description                        2
title                              0
image_thumbnail                 3112
dtype: int64

In [7]:
#cantidad de nulos en precio
datos.price.isnull().sum()

20410

In [8]:
#porcentaje de nulos en precio
100*datos.price.isnull().sum()/datos.price.shape[0]

16.837155584886982

In [9]:
#porcentaje de nulos en precio aproximado
100*datos.price_aprox_usd.isnull().sum()/datos.price_aprox_usd.shape[0]

16.837155584886982

In [10]:
#porcentaje de nulos en superficie total m2
100*datos.surface_total_in_m2.isnull().sum()/datos.surface_total_in_m2.shape[0]

32.44349117307375

In [11]:
#porcentaje nulos de precio por metro cuadrado
100*datos.price_usd_per_m2.isnull().sum()/datos.price.shape[0]

43.394654347467416

In [12]:
precio_m2_imputado=datos.apply(lambda x: x['price']/x['surface_total_in_m2']  if x['price_usd_per_m2'] is np.NAN else x['price_usd_per_m2'], axis=1)

#precio_m2 = data.apply(lambda x: x['price_usd_per_m2'] if x['Landsize'] == 0 else x['Price'] / x['Landsize'], axis=1)

#data['Precio_m2']=data.apply(lambda x: np.NaN if ((x['Landsize']  is np.NaN) | (x['Landsize']==0)) else x['Price']/x['Landsize'], axis=1)

precio_m2_imputado.isnull().sum()/precio_m2_imputado.shape[0]

0.43394654347467415

In [13]:
#Damos de bajo el currency
datos.currency.value_counts()

valor_de_cambio=datos.price_aprox_local_currency/datos.price_aprox_usd

serie_cambio=valor_de_cambio.value_counts()

print(serie_cambio)

mascara_cambio=serie_cambio==17.6445

print(serie_cambio.shape[0])

print(mascara_cambio.shape[0])

17.6445    82402
17.6445      714
17.6445      670
17.6445      224
17.6445      224
           ...  
17.6445        1
17.6445        1
17.6445        1
17.6445        1
17.6445        1
Length: 7238, dtype: int64
7238
7238


In [14]:
ubicacion = datos_crudos["place_with_parent_names"].str.split('|', expand=True) # Separo en varias columnas
ubicacion

0           1                             2              3  \
0          Argentina               Capital Federal      Mataderos   
1          Argentina        Bs.As. G.B.A. Zona Sur       La Plata   
2          Argentina               Capital Federal      Mataderos   
3          Argentina               Capital Federal        Liniers   
4          Argentina  Buenos Aires Costa Atlántica  Mar del Plata   
...    ..        ...                           ...            ...   
121215     Argentina               Capital Federal       Belgrano   
121216     Argentina      Bs.As. G.B.A. Zona Norte     San Isidro   
121217     Argentina               Capital Federal  Villa Urquiza   
121218     Argentina  Buenos Aires Costa Atlántica  Mar del Plata   
121219     Argentina               Capital Federal                  

                  4     5     6  
0                    None  None  
1                    None  None  
2                    None  None  
3                    None  None  
4            Centro        None  
...             ...   ...   ...  
121215               None  None  
121216       Beccar        None  
121217               None  None  
121218  Plaza Colón        None  
121219         None  None  None  

[121220 rows x 7 columns]

In [15]:

ubicacion.iloc[:,4].isna().sum()

4780

In [16]:
ubicacion_limpia=ubicacion.drop([0,1,6],axis=1,inplace=False)
ubicacion_limpia

2              3            4     5
0                    Capital Federal      Mataderos               None
1             Bs.As. G.B.A. Zona Sur       La Plata               None
2                    Capital Federal      Mataderos               None
3                    Capital Federal        Liniers               None
4       Buenos Aires Costa Atlántica  Mar del Plata       Centro      
...                              ...            ...          ...   ...
121215               Capital Federal       Belgrano               None
121216      Bs.As. G.B.A. Zona Norte     San Isidro       Beccar      
121217               Capital Federal  Villa Urquiza               None
121218  Buenos Aires Costa Atlántica  Mar del Plata  Plaza Colón      
121219               Capital Federal                        None  None

[121220 rows x 4 columns]

In [17]:
ubicacion_limpia.iloc[:,3].value_counts()


                             39892
Barrio Los Alisos              278
Barrio La Alameda               62
BarrioPortezuelo                54
Barrio Los Lagos                39
Barrio El Golf                  32
Enyoi                           17
Barrio Los Castores             14
QBay Yacht                      13
Barrio La Isla                  10
Barrio Las Glorietas            10
Islas del Canal                  6
Barrio Barrancas del Lago        4
Barrio Los Sauces                4
Barrio Cabos del Lago            3
Barrio Los Tilos                 1
Barrio El Yacht                  1
Name: 5, dtype: int64

In [18]:
ubicacion_limpia.iloc[:,1].value_counts()

Tigre             8983
Rosario           8504
Mar del Plata     7710
Córdoba           6606
                  4780
                  ... 
Playa Unión          1
Aldea Salto          1
Coronel Moldes       1
Coronel Bogado       1
Villa Unión          1
Name: 3, Length: 539, dtype: int64

In [19]:
ubicacion_limpia.iloc[:,2].value_counts()

                                  76023
Nordelta                           3863
La Plata                           1767
Olivos                             1590
Lomas de Zamora                    1253
                                  ...  
Barrio Privado El Rodal               1
Barrio Cerrado La Masia               1
Jeppener                              1
Barrio Cerrado "La Candelaria"        1
Princess                              1
Name: 4, Length: 568, dtype: int64

In [20]:
ubicacion_limpia.columns=['provincia','ciudad','barrio1','barrio2']

In [21]:
ubicacion_limpia

provincia         ciudad      barrio1 barrio2
0                    Capital Federal      Mataderos                 None
1             Bs.As. G.B.A. Zona Sur       La Plata                 None
2                    Capital Federal      Mataderos                 None
3                    Capital Federal        Liniers                 None
4       Buenos Aires Costa Atlántica  Mar del Plata       Centro        
...                              ...            ...          ...     ...
121215               Capital Federal       Belgrano                 None
121216      Bs.As. G.B.A. Zona Norte     San Isidro       Beccar        
121217               Capital Federal  Villa Urquiza                 None
121218  Buenos Aires Costa Atlántica  Mar del Plata  Plaza Colón        
121219               Capital Federal                        None    None

[121220 rows x 4 columns]

In [22]:
ubicacion_limpia['barrio1'].isna().sum()

4780

In [23]:
#aca pongo capturo los barrios de capital federal que estan en la columna ciudad
#esto se debe a que Capital Federal se guarda en la columna provinci

barrio_limpio=ubicacion_limpia.apply(lambda x: x['ciudad'] if x['provincia'] =='Capital Federal' else x['barrio1'],axis=1)

#precio_m2_imputado=datos.apply(lambda x: x['price']/x['surface_total_in_m2']  if x['price_usd_per_m2'] is np.NAN else x['price_usd_per_m2'], axis=1)
barrio_limpio

0             Mataderos
1                      
2             Mataderos
3               Liniers
4                Centro
              ...      
121215         Belgrano
121216           Beccar
121217    Villa Urquiza
121218      Plaza Colón
121219                 
Length: 121220, dtype: object

In [24]:
barrio_limpio.isna().sum()

3483

 LLEVANDO LOS BARRIOS DE CAPITAL BAJAMOS LA CANTIDAD EN BLANCO EN COLUMNA BARRIOS DE 4780 A 3483

In [25]:
ubicacion_limpia['barrio1']=barrio_limpio

In [26]:
ubicacion_limpia['barrio1'].value_counts()

                                      47499
Palermo                                4083
Nordelta                               3863
Belgrano                               2992
Caballito                              2273
                                      ...  
Terminal Nueva                            1
Barrio Melazzi                            1
El Zorzal                                 1
Barrio Cerrado El Casco de Alvarez        1
Princess                                  1
Name: barrio1, Length: 618, dtype: int64

AHORA LIMPIAMOS CIUDAD

In [27]:
ubicacion_limpia

provincia         ciudad        barrio1 barrio2
0                    Capital Federal      Mataderos      Mataderos    None
1             Bs.As. G.B.A. Zona Sur       La Plata                   None
2                    Capital Federal      Mataderos      Mataderos    None
3                    Capital Federal        Liniers        Liniers    None
4       Buenos Aires Costa Atlántica  Mar del Plata         Centro        
...                              ...            ...            ...     ...
121215               Capital Federal       Belgrano       Belgrano    None
121216      Bs.As. G.B.A. Zona Norte     San Isidro         Beccar        
121217               Capital Federal  Villa Urquiza  Villa Urquiza    None
121218  Buenos Aires Costa Atlántica  Mar del Plata    Plaza Colón        
121219               Capital Federal                                  None

[121220 rows x 4 columns]

In [28]:
ciudad_limpia=ubicacion_limpia.apply(lambda x: x['provincia'] if x['provincia'] =='Capital Federal' else x['ciudad'],axis=1)

In [29]:
ciudad_limpia

0         Capital Federal
1                La Plata
2         Capital Federal
3         Capital Federal
4           Mar del Plata
               ...       
121215    Capital Federal
121216         San Isidro
121217    Capital Federal
121218      Mar del Plata
121219    Capital Federal
Length: 121220, dtype: object

In [30]:
#ahora pisamos la ciudad vieja
ubicacion_limpia['ciudad']=ciudad_limpia

In [31]:
ubicacion_limpia

provincia           ciudad        barrio1 barrio2
0                    Capital Federal  Capital Federal      Mataderos    None
1             Bs.As. G.B.A. Zona Sur         La Plata                   None
2                    Capital Federal  Capital Federal      Mataderos    None
3                    Capital Federal  Capital Federal        Liniers    None
4       Buenos Aires Costa Atlántica    Mar del Plata         Centro        
...                              ...              ...            ...     ...
121215               Capital Federal  Capital Federal       Belgrano    None
121216      Bs.As. G.B.A. Zona Norte       San Isidro         Beccar        
121217               Capital Federal  Capital Federal  Villa Urquiza    None
121218  Buenos Aires Costa Atlántica    Mar del Plata    Plaza Colón        
121219               Capital Federal  Capital Federal                   None

[121220 rows x 4 columns]

In [32]:
ubicacion_limpia.sample(50)

provincia           ciudad  \
41165                       Santa Fe         Santa Fe   
61466       Bs.As. G.B.A. Zona Norte       San Miguel   
57792                Capital Federal  Capital Federal   
26631                Capital Federal  Capital Federal   
4848                 Capital Federal  Capital Federal   
45607   Buenos Aires Costa Atlántica          Pinamar   
45094                Capital Federal  Capital Federal   
89245                       Santa Fe          Rosario   
34972                        Córdoba          Córdoba   
108703                    Entre Ríos            Colón   
60968                Capital Federal  Capital Federal   
29877                        Córdoba          Córdoba   
112229               Capital Federal  Capital Federal   
45862                       Santa Fe          Rosario   
110871                      Santa Fe      Reconquista   
45079                       Santa Fe          Rosario   
50269                Capital Federal  Capital Federal   
121131               Capital Federal  Capital Federal   
48161       Bs.As. G.B.A. Zona Norte            Tigre   
25731                Capital Federal  Capital Federal   
116744               Capital Federal  Capital Federal   
109121                      Santa Fe          Rosario   
89188                        Formosa          Formosa   
14170   Buenos Aires Costa Atlántica    Mar del Plata   
107957      Bs.As. G.B.A. Zona Norte            Pilar   
76163                Capital Federal  Capital Federal   
13310       Bs.As. G.B.A. Zona Norte            Pilar   
112921      Bs.As. G.B.A. Zona Norte            Tigre   
88303                Capital Federal  Capital Federal   
103653               Capital Federal  Capital Federal   
65286         Bs.As. G.B.A. Zona Sur  Lomas de Zamora   
112295      Bs.As. G.B.A. Zona Norte            Tigre   
70400         Bs.As. G.B.A. Zona Sur         La Plata   
111851                      Santa Fe          Rosario   
70202         Bs.As. G.B.A. Zona Sur            Lanús   
46059                       Santa Fe          Rosario   
117283               Capital Federal  Capital Federal   
113845  Buenos Aires Costa Atlántica    Mar del Plata   
67069       Bs.As. G.B.A. Zona Oeste            Morón   
69686         Bs.As. G.B.A. Zona Sur       Avellaneda   
115656                       Córdoba          Córdoba   
1120        Bs.As. G.B.A. Zona Norte            Tigre   
5923          Bs.As. G.B.A. Zona Sur          Quilmes   
70059         Bs.As. G.B.A. Zona Sur         La Plata   
88848                        Córdoba                    
104261        Bs.As. G.B.A. Zona Sur  Almirante Brown   
4255        Bs.As. G.B.A. Zona Norte            Pilar   
5143                        Santa Fe      Castellanos   
69769         Bs.As. G.B.A. Zona Sur       Avellaneda   
2433        Bs.As. G.B.A. Zona Norte            Tigre   

                              barrio1 barrio2  
41165                                    None  
61466                           Muñiz          
57792                          Retiro    None  
26631                   San Cristobal    None  
4848                        San Telmo    None  
45607                                    None  
45094                     Parque Chas    None  
89245                                    None  
34972                                    None  
108703                                   None  
60968                         Palermo    None  
29877                                    None  
112229                        Palermo    None  
45862                                    None  
110871                                   None  
45079                                    None  
50269                     San Nicolás    None  
121131                       Recoleta    None  
48161                        Nordelta          
25731                           Boedo    None  
116744                       Belgrano    None  
109121                                   None  
89188          

estan mas limpio que antes , falta ver que hacer con la columna barrio2

In [33]:
ubicacion_limpia.barrio2.value_counts()

                             39892
Barrio Los Alisos              278
Barrio La Alameda               62
BarrioPortezuelo                54
Barrio Los Lagos                39
Barrio El Golf                  32
Enyoi                           17
Barrio Los Castores             14
QBay Yacht                      13
Barrio La Isla                  10
Barrio Las Glorietas            10
Islas del Canal                  6
Barrio Barrancas del Lago        4
Barrio Los Sauces                4
Barrio Cabos del Lago            3
Barrio Los Tilos                 1
Barrio El Yacht                  1
Name: barrio2, dtype: int64

In [34]:
#tiene 40 mil registros vacíos
#veamos que hay en aquellos registros que no estan vacios

mask_barrio2_vacia=ubicacion_limpia['barrio2']!='' 
mask_barrio2_vacia.value_counts()

True     81328
False    39892
Name: barrio2, dtype: int64

In [46]:
#Creo una mascara para filtrar los registros vacío y nulos
mascara_barrio2 = ((ubicacion_limpia['barrio2']!='') & (ubicacion_limpia['barrio2'].notnull()))

In [47]:
#Aplicamos el filtro a la columna barrio 2 y vemos cuantos valores quedan
barrio_2_filtrado=ubicacion_limpia['barrio2'].loc[mascara_barrio2]

In [48]:
barrio_2_filtrado.value_counts()

Barrio Los Alisos            278
Barrio La Alameda             62
BarrioPortezuelo              54
Barrio Los Lagos              39
Barrio El Golf                32
Enyoi                         17
Barrio Los Castores           14
QBay Yacht                    13
Barrio La Isla                10
Barrio Las Glorietas          10
Islas del Canal                6
Barrio Los Sauces              4
Barrio Barrancas del Lago      4
Barrio Cabos del Lago          3
Barrio El Yacht                1
Barrio Los Tilos               1
Name: barrio2, dtype: int64

In [49]:
#ahora veo que valores hay en barrio 1 cuando tengo algo en barrio2
bar1= ubicacion_limpia['barrio1'].loc[mascara_barrio2]
bar1.value_counts() 

Nordelta    548
Name: barrio1, dtype: int64

In [ ]:
ind=datos.columns
ind_label=np.arange(0,datos.shape[0]-1)

width = 0.35

fig = plt.figure(figsize=(16,6),edgecolor='gray'); ax = plt.axes()
#fig2= plt.figure(figsize=(16,6)); ax2 = plt.axes()

ax.barh(ind, valores_nulos, width, color='r') # barra roja. 
ax.barh(ind, valores_no_nulos, width, color='g',left=valores_nulos)#,right=valores_nulos)
#ax2.bar(ind, valores_no_nulos, width, color='b')#, bottom=valores_no_nulos) # barra azul 

ax.set_ylabel('Porcentaje'); ax.set_title('Relación Nulos y No-Nulos por Columna')
#ax.set_xticks(ind); 

#aca estaba como antes (ind)
#ax.set_xticklabels(ind_label)

#display(ind_label)

ax.tick_params(axis='both', which='minor', labelsize='small')

#ax.set_xticklabels(ind, rotation = 90, ha="right")
#ax2.set_xticklabels(ind,labelsize=8)
#ax.legend(labels=['Men', 'Women'])

ax.set_xticklabels(['0','20', '40', '60', '80', '100'])

#plt.gca().invert_xaxis()
plt.show()

#display(valores_nulos)

#display(valores_no_nulos)

#suma_ambos=valores_nulos+valores_no_nulos

#suma_ambos.value_counts()


In [1]:
#se puede usar scatterplot para ver el mapa
#sns.scatterplot(data = datos[datos["ciudad"] == "Capital Federal"].sample(1000), x="lon", y="lat",hue='property_type')

In [50]:
city1= ubicacion_limpia['ciudad'].loc[mascara_barrio2]
city1.value_counts() 

Tigre    548
Name: ciudad, dtype: int64

In [39]:
ubicacion_limpia['barrio2'].notnull()mask_bar2=ubicacion_limpia['barrio2'].notnull()
barrio_2_filtrado=ubicacion_limpia['barrio2'].loc[mask_bar2]


In [43]:
barrio_2_filtrado=barrio_2_filtrado.loc[mask_barrio2_vacia]

In [44]:
barrio_2_filtrado.value_counts()

Barrio Los Alisos            278
Barrio La Alameda             62
BarrioPortezuelo              54
Barrio Los Lagos              39
Barrio El Golf                32
Enyoi                         17
Barrio Los Castores           14
QBay Yacht                    13
Barrio La Isla                10
Barrio Las Glorietas          10
Islas del Canal                6
Barrio Los Sauces              4
Barrio Barrancas del Lago      4
Barrio Cabos del Lago          3
Barrio El Yacht                1
Barrio Los Tilos               1
Name: barrio2, dtype: int64

In [35]:
barrio_2_filtrado=ubicacion_limpia['barrio2'].loc[mask_barrio2_vacia]

In [36]:
barrio_2_filtrado.isna().sum()

80780

In [37]:

mask_barrio2_none=ubicacion_limpia['barrio2'] != pd.isna

mask_barrio2_none.value_counts()

True    121220
Name: barrio2, dtype: int64

avanzo con temas visualizacion